sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/DataLakeFileAppendOptions.java

```diff
@@ -83,4 +87,76 @@ public DataLakeFileAppendOptions setFlush(Boolean flush) {
         this.flush = flush;
         return this;
     }
+
+    /**
+     * Get lease action set on file.
+     * {@link LeaseAction#ACQUIRE} will attempt to acquire a new lease on the file,
+     * with {@link DataLakeFileAppendOptions#proposedLeaseId} as the lease ID.
+     * {@link LeaseAction#ACQUIRE_RELEASE} will attempt to aquire a new lease on the file,
+     * with {@link DataLakeFileAppendOptions#proposedLeaseId} as the lease ID. The lease will be released once the
+     * Append operation is complete.
+     * {@link LeaseAction#AUTO_RENEW} will attempt to renew the lease specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     * {@link LeaseAction#RELEASE} will attempt to release the least specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     *
+     * @return The {@link LeaseAction} set on the file.
+     */
+    public LeaseAction getLeaseAction() {
+        return leaseAction;
+    }
+
+    /**
+     * Get lease action set on file.
+     * {@link LeaseAction#ACQUIRE} will attempt to acquire a new lease on the file,
+     * with {@link DataLakeFileAppendOptions#proposedLeaseId} as the lease ID.
+     * {@link LeaseAction#ACQUIRE_RELEASE} will attempt to aquire a new lease on the file,
+     * with {@link DataLakeFileAppendOptions#proposedLeaseId} as the lease ID. The lease will be released once the
+     * Append operation is complete.
+     * {@link LeaseAction#AUTO_RENEW} will attempt to renew the lease specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     * {@link LeaseAction#RELEASE} will attempt to release the least specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     *
+     * @param leaseAction the {@link LeaseAction} to set on the file.
+     * @return the updated DataLakeFileAppendOptions object.
+     */
+    public DataLakeFileAppendOptions setLeaseAction(LeaseAction leaseAction) {
+        this.leaseAction = leaseAction;
+        return this;
+    }
+
+    /**
+     * @return the lease duration in seconds.
+     */
+    public Integer getLeaseDuration() {
```

Comment:
I'd rather see `Duration` used here to allow this to be more flexible on usage and avoid time as a number ambiguity (is it milliseconds, is it seconds, is it something else)

Reply:
I'm good with using `Integer` if we want to maintain internal consistency. In the future if we get feedback about it another overload using `Duration` could always be added, similar to what was done with `RequestRetryOptions` (https://github.com/Azure/azure-sdk-for-java/pull/17628).